sys/net/gnrc/application_layer/tftp/gnrc_tftp.c

```diff
@@ -1047,6 +1047,10 @@ int _tftp_decode_start(tftp_context_t *ctxt, uint8_t *buf, gnrc_pktsnip_t *outbu
 
     /* decode the TFTP transfer mode */
     for (uint32_t idx = 0; idx < ARRAY_LEN(_tftp_modes); ++idx) {
+        if (_tftp_modes[idx].len > (inpkt->size - sizeof(*hdr) - fnlen)) {
```

Comment:
according to the header format in [RFC](https://tools.ietf.org/html/rfc1350#appendix-I) all strings are 0 terminated so this can be simplified to (and also be more readable):

```
if (_tftp_modes[idx].len > strlen(str_mode) {
```

do you agree?

Reply:
No, I strongly disagree. Just because the RFC requires valid messages to use null-terminators for strings doesn't mean that an attacker can't forge a malicious message without terminating null bytes.

Since the packet buffer itself doesn't seem to be terminated with a null byte using a string function on it might result in an out-of-bounds read, reading memory of other packets in the pktbuf.

Reply:
okay sounds valid.